sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/BeamFnControlClient.java

```diff
@@ -150,15 +151,28 @@ public void processInstructionRequests(Executor executor)
   public BeamFnApi.InstructionResponse delegateOnInstructionRequestType(
       BeamFnApi.InstructionRequest value) {
     try {
-      return handlers.getOrDefault(value.getRequestCase(), this::missingHandler)
+      return handlers
+          .getOrDefault(value.getRequestCase(), this::missingHandler)
           .apply(value)
           .setInstructionId(value.getInstructionId())
           .build();
     } catch (Exception e) {
+      LOG.error(
+          "Exception while trying to handle {} {}",
+          BeamFnApi.InstructionRequest.class.getSimpleName(),
+          value.getInstructionId(),
+          e);
       return BeamFnApi.InstructionResponse.newBuilder()
           .setInstructionId(value.getInstructionId())
           .setError(getStackTraceAsString(e))
           .build();
+    } catch (Error e) {
+      LOG.error(
```

Comment:
Throwing the error here will just kill the thread within the executor and even if it was propagated by the executor, gRPC javadoc for onNext says we should call onError before throwing.

Should we:
* report the error back to the service?
* kill the process since errors are typically not recoverable (what to do with logs?)?
* attempt to continue?

Reply:
The error should be reported - 

Generally an Error is meant to be unrecoverable; so we have no guarantee of this call even working, but I think it's appropriate to propagate, at minimum, back to the runner harness.